### PR TITLE
HV-1137 Simplifying implementation of parameter and return value vali…

### PR DESCRIPTION
**Pull request by gunnarmorling**
…dation a bit

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java [hunk 1]
```diff
@@ -1173,32 +1171,29 @@ private int validatePropertyForNonDefaultGroup(ValueContext<?, Object> valueCont
 			while ( defaultGroupSequence.hasNext() ) {
 				Sequence sequence = defaultGroupSequence.next();
 				for ( GroupWithInheritance expandedGroup : sequence ) {
-					int numberOfViolationsOfCurrentGroup = 0;
+					int numberOfViolations = validationContext.getFailingConstraints().size();
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java]: Couldn't this statement be outside of the loop?

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java [hunk 2]
```diff
@@ -1387,33 +1378,29 @@ private int validatePropertyForNonDefaultGroup(ValueContext<?, Object> valueCont
...
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java]: Same here.

Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java]: Same here.

**Comment by gsmet** [comment]
See my comments inline.

I have mixed feeling about this. It does not really simplify the logic and we lose an information (which is admittedly useless at the time). No strong opinion though so if you prefer it like that, let's go for it.

**Comment by gunnarmorling** [comment]
Reworked the commit to address your remark and merged. Thanks for reviewing, @gsmet!

**Pull request closed without merging**